provider/pprof/index.ts

```diff
@@ -0,0 +1,89 @@
+import { dirname } from 'path'
+import type {
+    Annotation,
+    AnnotationsParams,
+    AnnotationsResult,
+    MetaParams,
+    MetaResult,
+    Provider,
+    ProviderSettings,
+} from '@openctx/provider'
+import { escapeSpecial, parseGolang } from './parser.js'
+import { type Node, findReportPath, getPprof } from './pprof.js'
+
+/**
+ * An [OpenCtx](https://openctx.org) provider that annotates every function declaration with
+ * the CPU time and memory allocations associated with it.
+ */
+const pprof: Provider = {
+    meta(params: MetaParams, settings: ProviderSettings): MetaResult {
+        return {
+            name: 'pprof',
+            annotations: {
+                selectors: [{ path: '**/*.go' }],
+            },
+        }
+    },
+
+    annotations(params: AnnotationsParams, settings: ProviderSettings): AnnotationsResult {
+        // TODO: check it is not a test file, skip otherwise
+
+        const pprof = getPprof()
+        if (pprof === null) {
+            // TODO: log that no command line tool was found. Ideally, do it once on init.
+            return []
+        }
+
+        // let workspaceRoot: string | undefined = undefined
+        // const workspaceFolder = vscode.workspace.getWorkspaceFolder(vscode.Uri.parse(params.uri))
+        // if (workspaceFolder && workspaceFolder !== null) {
+        //     workspaceRoot = workspaceFolder.uri.path
+        // }
```

Comment:
I see you have this commented out, but you still have an incantation in your esbuild which shows you want to do this. Did you have trouble getting this to work?